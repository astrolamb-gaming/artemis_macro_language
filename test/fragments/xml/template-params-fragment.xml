<mission_data version="1.0">
  <templates>
    <template name="tsn:GM_Buttons">
			<params>
				<param name="x_across"/>
				<param name="y_across"/>
				<param name="x_start"/>
				<param name="y_start"/>
				<param name="x_gap"/>
				<param name="y_gap"/>
				<param name="h"/>
				<param name="w"/>
			</params>
			<repeat _length="${y_across}" _as="y"> 
				<repeat _length="${x_across}" _as="x">
					<set_gm_button text="Sector ${_.int(y)*_.int(x_across)+_.int(x)+1}" x="${_.int(w)*_.int(x)+_.int(x_gap)*_.int(x)+_.int(x_start)}" y="${_.int(y) +_.int(y_start)}" h="${h}" w="${w}" />
				</repeat>
			</repeat>
		</template>
    </templates>
    
  <start>
	<tsn:GM_Buttons x_across="5" y_across="2" x_start="100" y_start="200" x_gap="3" y_gap="2" h="10" w="100" />
  </start>
</mission_data>
