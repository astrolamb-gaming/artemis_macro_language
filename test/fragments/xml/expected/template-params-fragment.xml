<mission_data version="1.0">
  <start>
    <set_gm_button text="Sector 1" x="100" y="200" h="10" w="100"/>
    <set_gm_button text="Sector 2" x="203" y="200" h="10" w="100"/>
    <set_gm_button text="Sector 3" x="306" y="200" h="10" w="100"/>
    <set_gm_button text="Sector 4" x="409" y="200" h="10" w="100"/>
    <set_gm_button text="Sector 5" x="512" y="200" h="10" w="100"/>
    <set_gm_button text="Sector 6" x="100" y="201" h="10" w="100"/>
    <set_gm_button text="Sector 7" x="203" y="201" h="10" w="100"/>
    <set_gm_button text="Sector 8" x="306" y="201" h="10" w="100"/>
    <set_gm_button text="Sector 9" x="409" y="201" h="10" w="100"/>
    <set_gm_button text="Sector 10" x="512" y="201" h="10" w="100"/>
  </start>
</mission_data>